okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -296,6 +342,8 @@ public void start(Listener listener) {
         source = Okio.buffer(Okio.source(socket));
         sink = Okio.buffer(Okio.sink(socket));
       } catch (IOException e) {
+        // TODO(jhump): should we instead notify the listener of shutdown+terminated?
+        // (and probably do all of this work asynchronously instead of in calling thread)
```

Comment:
@madongfly and I recently dealt with this exact exception very recently in #409.

Yes, ideally it would be async. Both throwing an exception and calling shutdown+terminated have their own gotchas: backing-out changes for the exception, reentrancy bugs for the shutdown+terminated.